Assets/Tools/DeveloperConsole.cs

```diff
@@ -0,0 +1,389 @@
+        if (newStatus) // if it opens the console, focus the console and clear it
+        {
+            FocusConsole();
+            ClearConsole();
+
+            Cursor.lockState = CursorLockMode.None;
+            Cursor.visible = true;
+            Time.timeScale = 0;
+        }
+        else
+        {
+            Cursor.lockState = CursorLockMode.Locked;
+            Cursor.visible = false;
+            Time.timeScale = 1;
+        }
+    }
+    #endregion // Console State
+
+    #region Read Input and Update Console
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    public void ReadStringInput(string s)
+    {
+        input = s;
+        inputModified = input.Split(new string[] { " " }, StringSplitOptions.RemoveEmptyEntries); // to split up the command from the parameters
+        userInput.text = "";
+
+        // executes the command and focuses the input field
+        ExecuteCommand();
+        FocusConsole();
+    }
+
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    private void AddToConsoleLog(string message)
+    {
+        consoleLogText += "> " + message + "\n";
+        consoleLogUI.SetText(consoleLogText);
+        UpdateConsoleLog();
+    }
+
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    public void ClearConsole()
+    {
+        consoleLogText = "";
+        UpdateConsoleLog();
+    }
+
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    /// Updates the console log to match new text
+    /// 
+    private void UpdateConsoleLog()
+    {
+        consoleLogUI.SetText(consoleLogText);
+    }
+    #endregion // Read Input and Update Console
+
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    /// Checks if command inputted exists
+    /// 
+    bool CheckCommand(string[] command, out DevCommand commandType) // checks if the command inputted exists
+    {
+        if (command.Length != 0) // makes sure an empty command wasn't sent
+        {
+            if (Enum.TryParse<DevCommand>(command[0], out commandType)) return true;
+            else
+            {
+                AddToConsoleLog("Error: Command \"" + input + "\" doesn't exist!");
+                return false;
+            }
+        }
+        commandType = DevCommand.error;
+        return false;
+    }
+
+    ///-/////////////////////////////////////////////////////////////////////////////
+    ///
+    private void ExecuteCommand()
+    {
+        DevCommand commandType;
+        if (CheckCommand(inputModified, out commandType)) // checks if the command exists
+        {
+            //---------------------
+            // EXECUTE THE COMMANDS
+            //---------------------
+            if (commandType == DevCommand.help) // checks for help command which shows documentation
```

Comment:
Can you use `switch` statement instead of all the `if() else if()` for checking `commandType` enum?

Reply:
Fixed!